### ISSUE TITLE
HDMI output frequency

### ISSUE THREAD

**Issue by omski** [mrfixit2001/recalbox_rockpi4#6]
I could not connect to my Dell P2417H because of the requested display res/freq.
Please add support for 1920P @60Hz - Thanks!

**Comment by omski** [comment]
Used a TV (HDMI) and got sound but still no video output.
Board is OK - used Armbian with no issues.

I used the RockPi 4 image which is said to run an NanoPi M4 also (https://www.youtube.com/watch?v=rgj9mR6Pl04)
Hardware:
Nanopi M4, 4Gb version
5V @PSU
tried multiple SD cards (SanDisk Ultras and Extreme)
reloaded the image too rule out file errors

**Comment by omski** [comment]
Logged in via SSH and the output from tvservice -s
states that it runs in 1080p 60Hz as expected. Sound is running and it seems that the system is running. For some reason there is just no video output - just a black screen. The TV shows that there is a HDMI signal but there seems to be some output missing. Any idea what I can do about it? Is there a way to diag the gfx path. Thanks!

**Comment by StuartIanNaylor** [comment]
See if the same as my Armbian images that seem to do the same. Let it boot disconnect and plug back in.
Hey presto hdmi again but dunno what it is.

```
1:.ROCKPI4
Retrieving file: /Image
reading /Image
21739528 bytes read in 3445 ms (6 MiB/s)
append: console=tty3 rw root=/dev/mmcblk0p2 rootwait rootfstype=ext4 panic=10 init=/sbin/init coherent_pool=1M ethaddr=86:4e:59:ee:ba:aa eth1addr= serial=98d44114ec0a8ad cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory swapaccount=1 video=HDMI-A-1:1920x1080@60 quiet loglevel=3 vga=current
Retrieving file: /rockpi-4b-linux.dtb
reading /rockpi-4b-linux.dtb
92695 bytes read in 19 ms (4.7 MiB/s)
## Flattened Device Tree blob at 01f00000
   Booting using the fdt blob at 0x1f00000
   Loading Device Tree to 000000007deea000, end 000000007df03a16 ... OK

Starting kernel ...

[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Initializing cgroup subsys cpuset
[    0.000000] Initializing cgroup subsys cpu
[    0.000000] Initializing cgroup subsys cpuacct
[    0.000000] Linux version 4.4.169 (adam@adam-vb) (gcc version 7.1.1 20170707 (Linaro GCC 7.1-2017.08) ) #3 SMP Thu Jan 10 20:05:09 EST 2019
[    0.000000] Boot CPU: AArch64 Processor [410fd034]
[    0.000000] earlycon: Early serial console at MMIO32 0xff1a0000 (options '')
[    0.000000] bootconsole [uart0] enabled
```
Latest version doesn't seem to boot though

**Comment by omski** [comment]
I am pretty sure that I left enough time for a full boot and did several retries without getting any video output but I will retry again. In regards to your Armbian experience - the latest Armbian image does not boot here either. It goes into continuous boot loop. Some images can be 'fixed' by exchanging the .dtb - I extracted mine from a working version of librelec. I am using a Friendlyelec NanoPi M4 (4Gb version) and it seems that the RockPi 4 is compatible in most regards.

**Comment by omski** [comment]
Used the rockpro64 image (190603) with dtb rockpi4 from librelec and now I can see video output. Hope this helps...

**Comment by omski** [comment]
> See if the same as my Armbian images that seem to do the same. Let it boot disconnect and plug back in.

@StuartIanNaylor 
I have tried Armbian_5.90_Nanopim4_Ubuntu_bionic_default_4.4.179 again and switched through all dtb  files contained in /boot/dtb in /boot/armbianEnc.txt 
NO success
BUT as I copied the rk3399-rock-pi-4.dtb from the before mentioned librelec distribution into the dtb directory and referenced it in the armbianEnc.txt VIOLA armbian booted!
I will try the reverse and copy the dtb files from armbian to librelec...

Has anyone an idea what could be wrong? Even the stock images from friendlyelec fail to boot. But did not find an dtb files to fiddle around with. Could it be that there are hardware differences which prevent some device tree definitions to work with certain devices?
